backend/apps/claims/services/claim_service.py

```
@@ -20,57 +21,55 @@ def create_claim(*, data: dict[str, Any], user=None) -> Claim:
             from apps.members.models import Person
             if isinstance(member_id, str):
                 try:
-                    member = Person.objects.get(id=member_id)
+                    member = Person.objects.get(id=member_id, is_deleted=False)
                     data["member"] = member
                 except Person.DoesNotExist:
-                    raise ValidationError("Invalid member ID")
+                    raise NotFoundError("Person", member_id)
```

Ruff (0.14.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)